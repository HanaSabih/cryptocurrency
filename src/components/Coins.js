import React, { useEffect } from 'react';
import { useSelector, useDispatch } from 'react-redux';
import { coinList, coinFilter } from '../redux/coins/coinsSlice';
import Coin from './Coin';

const Coins = () => {
  const dispatch = useDispatch();
  const crytoList = useSelector((store) => store.cryto);
  useEffect(() => {
    dispatch(coinList());
  });
  const changeHandler = (e) => {
    dispatch(coinFilter(e.target.value));
  };

  const searchHandler = (e) => {
    e.preventDefault();
    window.location.href = `/detail/${e.target.coinSearch.value}`;
  };

  return (
    <section>
      <div className="infos d-none">
        <select className="filter" onChange={changeHandler}>
          <option value="price"> Rank </option>
          <option value="name"> Name </option>
          <option value="vol"> Price </option>
        </select>
        <form onSubmit={searchHandler}>
          <input type="text" name="coinSearch" placeholder="Crypto cryto name" required />
          <input type="submit" value="Search" />
        </form>
      </div>
      <div>
        <ul className="d-grid mx-0 px-0 cardContainer">
          {crytoList.map((cryto) => (
            <Coin
              key={cryto.id}
              id={cryto.id}
              symbol={cryto.symbol}
              price={cryto.priceUsd.substring(0, 8)}
            />
          ))}
        </ul>
      </div>
    </section>
  );
};

export default Coins;
