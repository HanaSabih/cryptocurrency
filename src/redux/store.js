import {
  combineReducers,
} from 'redux';
import { configureStore } from '@reduxjs/toolkit';
import cryto from './coins/coinsSlice';
import detail from './details/detailsSlice';

const reducer = combineReducers({
  cryto,
  detail,
});

const store = configureStore({
  reducer,
});

export default store;
