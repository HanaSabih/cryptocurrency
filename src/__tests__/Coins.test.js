import { Provider } from 'react-redux';
import renderer from 'react-test-renderer';
import store from '../redux/store';
import Coins from '../components/Coins';

describe('Coins component', () => {
  test('component match with the snapshot', () => {
    const tree = renderer.create(
      <Provider store={store}>
        <Coins />
      </Provider>,
    ).toJSON();
    expect(tree).toMatchSnapshot();
  });
});
