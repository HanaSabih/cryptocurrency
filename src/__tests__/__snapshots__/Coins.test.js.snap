// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Coins component component match with the snapshot 1`] = `
<section>
  <div
    className="infos d-none"
  >
    <select
      className="filter"
      onChange={[Function]}
    >
      <option
        value="price"
      >
         Rank 
      </option>
      <option
        value="name"
      >
         Name 
      </option>
      <option
        value="vol"
      >
         Price 
      </option>
    </select>
    <form
      onSubmit={[Function]}
    >
      <input
        name="coinSearch"
        placeholder="Crypto cryto name"
        required={true}
        type="text"
      />
      <input
        type="submit"
        value="Search"
      />
    </form>
  </div>
  <div>
    <ul
      className="d-grid mx-0 px-0 cardContainer"
    />
  </div>
</section>
`;
